src/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -523,7 +525,7 @@ def _format_checkpoint_name(
             name = group[1:]
 
             if auto_insert_metric_name:
-                filename = filename.replace(group, name + "={" + name)
+                filename = filename.replace(group, name + cls.CHECKPOINT_EQUALS_CHAR + "{" + name)
```

Comment:
@carmocca's suggestion was to also replace cls here with self (removing the `@classmethod` decorator and making it a regular method. Could be done here or in a separate PR if you like.

Reply:
My only doubt is that by doing so I'm not sure of what's the point of keeping those values as class attributes instead of normal instance attributes, so I had proposed a separate PR to talk about the possibility of changing that as well.
But if you already know how you would like it best I can just change it to that right now